frontend/app/monitoring/HealthMonitoringTask.scala

```diff
@@ -0,0 +1,124 @@
+package monitoring
+
+import java.io.File
+import java.lang.management.ManagementFactory
+
+import akka.actor.ActorSystem
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.services.cloudwatch.model._
+import com.amazonaws.util.EC2MetadataUtils
+import com.gu.monitoring.CloudWatch.cloudwatch
+import com.gu.monitoring.CloudWatchHealth
+import com.typesafe.scalalogging.StrictLogging
+import monitoring.CloudwatchMetric.MetricItem
+import monitoring.Scheduler.TaskDefinition
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration.{FiniteDuration, _}
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.{Failure, Success}
+
+object HealthMonitoringTask extends StrictLogging {
+
+  def start(implicit system: ActorSystem, executionContext: ExecutionContext, stage: String, appName: String): Unit = {
+
+    val dimensions: Seq[Dimension] = Seq(
+      new Dimension().withName("App").withValue(appName),
+      new Dimension().withName("Stage").withValue(stage),
+      new Dimension().withName("Services").withValue("jvm")
+    ) ++
+      Option(EC2MetadataUtils.getInstanceId).map { instanceId =>
+        new Dimension().withName("InstanceId").withValue(instanceId)
+      }.toList
+
+    val metricDefinitions = Seq[MetricItem](
+      MetricItem("max-heap-memory", ManagementFactory.getMemoryMXBean.getHeapMemoryUsage.getMax, StandardUnit.Bytes),
+      MetricItem("used-heap-memory", ManagementFactory.getMemoryMXBean.getHeapMemoryUsage.getUsed, StandardUnit.Bytes),
+      MetricItem("committed-heap-memory", ManagementFactory.getMemoryMXBean.getHeapMemoryUsage.getCommitted, StandardUnit.Bytes),
+      MetricItem("max-non-heap-memory", ManagementFactory.getMemoryMXBean.getNonHeapMemoryUsage.getMax, StandardUnit.Bytes),
+      MetricItem("used-non-heap-memory", ManagementFactory.getMemoryMXBean.getNonHeapMemoryUsage.getUsed, StandardUnit.Bytes),
+      MetricItem("committed-non-heap-memory", ManagementFactory.getMemoryMXBean.getNonHeapMemoryUsage.getCommitted, StandardUnit.Bytes),
+
+      MetricItem("free-disk-space", new File("/").getUsableSpace, StandardUnit.Bytes),
+
+      MetricItem("thread-count", ManagementFactory.getThreadMXBean.getThreadCount, StandardUnit.Count)
+    ) ++
+      ManagementFactory.getGarbageCollectorMXBeans.asScala.flatMap { gcBean =>
+        val name = gcBean.getName.replace(" ", "_")
+        Seq(
+          MetricItem(s"$name-gc-count-per-min", gcBean.getCollectionCount, StandardUnit.Count),
+          MetricItem(s"$name-gc-time-per-min", gcBean.getCollectionTime, StandardUnit.Milliseconds)
+        )
+      }
+
+    Scheduler.schedule(
+      task = TaskDefinition(
+        name = "HealthMonitoringTask",
+        task = () => CloudwatchMetric.putMetrics(dimensions, metricDefinitions)),
+      initialDelay = 1.second,
+      interval = 10.seconds
+    )
+  }
+
+}
+
+// simple stateless fire and forget scheduler
+object Scheduler extends StrictLogging {
```

Comment:
looks like it was always a trait, but no explanation why https://github.com/guardian/membership-frontend/blob/b9e12c6bd2e2b7e5f6642d3e0bd1b92e05a401ff/frontend/app/utils/ScheduledTask.scala